### PR TITLE
Library_Reflection_Versioning_Engine: Update BHoM folder references

### LINKED ISSUE #1822: BHoM_Engine: Update any references to appdata BHoM folder in the code
#### Parent issue(s) or Pull Request(s) causing the changes:

https://github.com/BHoM/admin/issues/5

#### Description:

All references to the BHoM roaming folder should be re-pointed to the new ProgramData folder for BHoM as discussed in admin issue.

This affects:
Library_Engine
Versioning_Engine
Reflection_Engine

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #1822 

Update references pointing at the roaming folder to be pointing at the ProgramData folder instead.

All done in support of https://github.com/BHoM/admin/issues/5

### CODE REVIEW

**Comment by FraserGreenroyd** [comment]
Just for double safety (as I know this is a draft) I have added the `do-not-merge` tag as a double safe reminder :smile:

File: Library_Engine/Query/Libraries.cs [hunk 1]
```diff
@@ -228,7 +228,7 @@ private static void AddToPathDictionary(string path, string dictionaryPath)
         /**** Private Fields                            ****/
         /***************************************************/
 
-        private static readonly string m_sourceFolder = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), @"BHoM\DataSets");
+        private static readonly string m_sourceFolder = @"C:\ProgramData\BHoM\DataSets";
```
Comment (al-fisher) [inline Library_Engine/Query/Libraries.cs:231]:
```suggestion
        private static readonly string m_sourceFolder = @"C:\ProgramData\BHoM\Datasets";
```

Hey @IsakNaslundBh - can we make this `Datasets` here? Where else would this need a change?

I was just posting on the default post-build events for toolkits - as well as specific for BHoM.csproj which can assist in making directories so good time to confirm

Comment (al-fisher) [reply Library_Engine/Query/Libraries.cs:231]: Happy to leave as is if painful as part of this PR as discussed with regards to other tweaks

Comment (IsakNaslundBh) [reply Library_Engine/Query/Libraries.cs]: update done. Agree with this change, as long, as you mentioned, we track it through in all postbuilds :)

**Pull request merged**